scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/GetCreatedActivitiesAction.java

```diff
@@ -11,15 +11,18 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 /*package-local*/ final class GetCreatedActivitiesAction implements AutoCloseable {
   private static final @Language("SQL") String sql = """
     select
-      c.goal_id,
-      c.goal_revision,
+      a.goal_id,
+      a.goal_revision,
+      c.goal_invocation_id,
       c.activity_id
     from scheduler.scheduling_goal_analysis_created_activities as c
-    where c.analysis_id = ?
+    join scheduler.scheduling_goal_analysis as a using (goal_invocation_id)
+    where c.analysis_id = ? and a.analysis_id = ?
```

Comment:
The need for the second parameter can be removed by adding `analysis_id` to the `using` clause:
```suggestion
    join scheduler.scheduling_goal_analysis as a using (analysis_id, goal_invocation_id)
    where c.analysis_id = ?;
```

This comment also applies to `GetSatisfyingActivitiesAction`

Reply:
These are functionally equivalent right?

Reply:
Technically not, as the suggestion should create a smaller join table while yours has a bigger table that then gets filtered down more, but it should return the same result set and I don't know enough about postgres query optimization to know the performance difference.

Honestly I was going to suggest that the `and` clause gets replaced by `c.analysis_id = a.analysis_id` until I realized that I was just defining a `join` condition. The big takeaway about this comment is that there's no need to add a second parameter to this query that is the same value as the other parameter. (Additionally, only passing in the value once enforces that both `analysis_id`s are the same, while _technically_ it's just happenstance that in the only usage of the query they are the same)